WebApp-B2C-DotNet/App_Start/Startup.Auth.cs

```diff
@@ -44,59 +38,54 @@ public void ConfigureAuth(IAppBuilder app)
 
             app.UseCookieAuthentication(new CookieAuthenticationOptions());
 
-            OpenIdConnectAuthenticationOptions options = new OpenIdConnectAuthenticationOptions
+            // Configure OpenID Connect middleware for each policy
+            app.UseOpenIdConnectAuthentication(CreateOptionsFromPolicy(SignUpPolicyId));
+            app.UseOpenIdConnectAuthentication(CreateOptionsFromPolicy(ProfilePolicyId));
+            app.UseOpenIdConnectAuthentication(CreateOptionsFromPolicy(SignInPolicyId));
+        }
+
+        // Used for avoiding yellow-screen-of-death
+        private Task AuthenticationFailed(AuthenticationFailedNotification<OpenIdConnectMessage, OpenIdConnectAuthenticationOptions> notification)
+        {
+            notification.HandleResponse();
+            if (notification.Exception.Message == "access_denied")
+            {
+                notification.Response.Redirect("/");
```

Comment:
What's the user experience if the auth fails for access_denied? Do they just get redirected to the root without any error message?